bsb_hdf5/placement_set.py

```diff
@@ -197,18 +203,18 @@ def load_additional(self, key=None, handle=HANDLED):
     @handles_handles("r")
     def _get_morphology_loaders(self, handle=HANDLED):
         stor_mor = {}
+        meta = self._engine.morphologies.get_all_meta()
         for chunk in self.get_loaded_chunks():
             path = self.get_chunk_path(chunk)
             try:
                 _map = handle[path].attrs["morphology_loaders"]
             except KeyError:
                 continue
-            stor_mor.update(
-                (m.name, m)
-                for m in self._engine.morphologies.select(
-                    _MapSelector(ps=self, names=_map)
-                )
-            )
+            for label in _map:
+                if label not in stor_mor and label in meta:
```

Comment:
why `and label in meta`? Wouldn't this exclude morphologies without metadata from being loaded?

If metadata is required I think it's better to raise an error if there's no metadata, than to quietly not load the morphology. This is a possible source of confusion for users.

Reply:
(great optimization btw!)

Reply:
Added a test to catch this case.